module/src/main/java/org/mule/extension/spring/internal/beanfactory/ArtifactObjectsAwareBeanFactory.java

```
@@ -84,6 +109,12 @@ public Object doResolveDependency(DependencyDescriptor descriptor, String beanNa
    */
   @Override
   protected <T> T doGetBean(String name, Class<T> requiredType, Object[] args, boolean typeCheckOnly) throws BeansException {
+    if ("fips140-2".equals(System.getProperty("mule.security.model"))) {
+      if (name.contains(DaoAuthenticationProvider.class.getName()) || DaoAuthenticationProvider.class.equals(requiredType)) {
+        UserDetailsService userDetailsService = (UserDetailsService) super.getBean("userService");
```

I am not sure about this hardcoded name, where this bean name come from?

---

When the user defines the authenticationManager in beans.xml [like this](https://github.com/mulesoft/mule-spring-module/blob/47728877ba2765221203a7ad4042db80b4b575b0/module/src/test/resources/beans.xml), they provide an ID for user-service. I haven't found a way to parameterise this, which could be a limitation for customers who need to use a different name for the ID instead of userService.

---

Please tell me if these statements are correct:

1. If we are running in a non-FIPS environment, the user is free to configure the bean with the name he wants
2. In FIPS environments, we need to configure that bean and that is why we are expecting it with a specific name.

Now my questions:
1. Does the name that the method receives correspond to the name of that same bean?
2. Is it not possible to search for the bean by type and in the case that there is only one, use it independently of the name?

I am not convinced by the hardcoded name....

---

Yes, the above depiction is correct. In a non-FIPS environment, the customer is free to name it anything, but the same does not apply in FIPS.

1. The method receives the name org.springframework.security.authentication.dao.DaoAuthenticationProvider, which is different.
2. I tried searching by type, and it works fine when only one bean is configured. However, it throws an error if multiple beans are configured. Even if we get a list of bean names for a UserDetailService type, we can never be certain which one to use.

---

@elcoya , what do you think about this? Should we try searching for the bean by type if only one exists, and if not, default to a hardcoded name?

Should we use a fixed hardcoded name, or should we define a default one and allow it to be modified in some way (e.g., through parameterization or a system property)?

---

@mule-niraj if possible, this constant value (now hardcoded) should be override-able. Is there any scenario where we can turn this into a value that is resolved based on a more flexible criteria? Can we have this as default but allow for a system property to define another?

---

Just to make the idea a little more concrete, leave this value as the default, but allow for an override with a system property. For the time being, we don't even need to document this system property, but let's plan for contingencies.

---

Hi, @mule-niraj ,
I’ve been discussing this with @elcoya, and we believe the best approach is to have a default name (userService) while also allowing it to be overridden via a system property.
For now, we can leave this out of the documentation—at least until a customer requests this capability.
Could you add the system property and include a test to cover it?
Thanks!

---

@acurci-at-mulesoft and @elcoya, I've continued debugging the code and found a solution where we no longer need to use the system property to override the userService name. Instead, we can use `getBeanDefinition()` to retrieve the `DaoAuthenticationProvider` instance and access the userDetailsService from the `propertyValue` list provided by the user. I've implemented this change in the code. Could you both please review it? Thanks!